.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -16,3 +16,8 @@
 [If this is a bug-fix or enhancement, it closes issue # ]
 [If this is a new feature, it implements the following paper: ]
 
+## For reviewers
+
+- [ ] Check that the PR title is short, concise, and will make sense 1 year
+  later.
+- [ ] Check that new features are mentioned in `doc/release/release_dev.rst`.
```

Comment:
I think that if you do it right it'll work great. "Hey X! This is such a cool new feature, do you want to add it to `release_dev` so that it gets highlighted next release?"

Otherwise we might be able to use the editable branch feature, or do a fixup PR quickly following the merge.